server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -1699,7 +1700,18 @@ public synchronized void close() throws IOException {
      * logic should use Node Shutdown, see {@link org.elasticsearch.cluster.metadata.NodesShutdownMetadata}.
      */
     public void prepareForClose() {
+        ThreadPool threadPool = injector.getInstance(ThreadPool.class);
+        HttpServerTransport httpServerTransport = injector.getInstance(HttpServerTransport.class);
+
+        Future<?> httpServerStopped = threadPool.generic().submit(httpServerTransport::stop);
```

Comment:
I think we can do this in a dedicated thread created here. We don't need to balance the work of this thread compared to other work, and this isn't a generic task in the system, it is special for shutdown.

Reply:
Changed to using a newly created thread.